stl/inc/random

```diff
@@ -2779,19 +2779,24 @@ public:
             return _Mean;
         }
 
+        static constexpr double _Small_threshold = 12.0;
+
         void _Init(_Ty1 _Mean0) noexcept { // set internal state
             _STL_ASSERT(0.0 < _Mean0, "invalid mean argument for poisson_distribution");
             _Mean = _Mean0;
+            if (_Mean0 < _Small_threshold) {
+                _Small._Init(_Mean0);
+                return;
+            }
             _Sqrt = _CSTD sqrt(2.0 * _Mean0);
             _Logm = _CSTD log(_Mean0);
             _Gx1  = _Mean0 * _Logm - _XLgamma(_Mean0 + 1.0);
-            _Small._Init(_Mean0);
         }
 
-        _Ty1 _Mean;
-        _Ty1 _Sqrt;
-        _Ty1 _Logm;
-        _Ty1 _Gx1;
+        _Ty1 _Mean = 0.0;
+        _Ty1 _Sqrt = 0.0;
+        _Ty1 _Logm = 0.0;
+        _Ty1 _Gx1  = 0.0;

```

Comment:
No change requested. Now it seems clear that the member group {`_Sqrt`, `_Logm`, `_Gx1`} and the `_Small` member never logically coexist. So perhaps we can store them into a union with `_Mean` being the tag, which makes `param_type` smaller. Although such change is ABI breaking and can only be done since vNext.